pubtools/_quay/operator_pusher.py

```diff
@@ -547,39 +547,75 @@ def build_index_images(self):
             is_advisory_source = all(
                 [re.match(r"^[A-Z0-9:\-]{4,40}$", item.origin) for item in non_fbc_items]
             )
-            item_groups = {}
+            item_groups = {
+                version: {
+                    "items": [],
+                    "overwrite": True,
+                    "build_tags": [],
+                    "destination_tags": [version],
+                }
+            }
             if is_hotfix and not is_advisory_source:
                 raise ValueError("Cannot push hotfixes without an advisory")
+
             if is_hotfix:
                 for item in non_fbc_items:
-                    item_groups.setdefault(item.origin, []).append(item)
+                    hotfix_tag = "{0}-{1}-{2}".format(
+                        version,
+                        item.metadata["com.redhat.hotfix"],
+                        item.origin.split("-")[1].replace(":", "-"),
+                    )
+                    item_groups.setdefault(
+                        item.origin,
+                        {
+                            "items": [],
+                            "overwrite": False,
+                            "build_tags": [],
+                            "destination_tags": [hotfix_tag],
+                        },
+                    )
+                    item_groups[item.origin]["items"].append(item)
             else:
-                item_groups["default"] = non_fbc_items
+                for item in non_fbc_items:
+                    if item.metadata.get("com.redhat.pre-release"):
+                        item_groups.setdefault(
+                            f'{version}.{item.metadata.get("com.redhat.pre-release")}',
```

Comment:
can you please share why you are using f'{version}.{item.metadata.get("com.redhat.pre-release")}' here, but for hotfix on line 569, item.origin is used

Reply:
because for hotfix origin is set to advisory, for pre-release content origin can be anything